src/fermilib/transforms/_bksf.py

```diff
@@ -0,0 +1,384 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""BKSF transform on fermionic operators."""
+#from __future__ import absolute_import
+
+import numpy as np
+from fermilib.ops import InteractionOperator
+from projectq.ops import QubitOperator
+import networkx as nx
+
+
+def bksf(operator):
+    """ Find the bksf pauli-representation of InteractionOperator. 
+        bksf pauli-representation of general FermionOperator is 
+        not possible in bksf. So the InteractionOperator given as input must
+        be Hermitian.
+    Returns:
+        transformed_operator: An instance of the QubitOperator class.
+
+    """
+    if isinstance(operator, InteractionOperator):
+        return bksf_interaction_op(operator)
+    else:
+        TypeError("operator must be an InteractionOperator.")
+  
+
+def bksf_interaction_op(iop, n_qubits=None):
+    """Output InteractionOperator as QubitOperator class under BKSF transform.
+
+    Input: Interaction operator to be transformed.
+    Returns:
+        qubit_operator: An instance of the QubitOperator class.
+    """
+    from fermilib.utils import count_qubits
+    if n_qubits is None:
+        n_qubits = count_qubits(iop)
+    if n_qubits < count_qubits(iop):
+        n_qubits = count_qubits(iop)
+
+    # Initialize qubit operator as constant.
+    qubit_operator = QubitOperator((), iop.constant)
+    edge_matrix = bksf_edge_matrix(iop)
+    # Loop through all indices.
+    for p in range(n_qubits):
+        for q in range(n_qubits):
+
+            # Handle one-body terms.
+            coefficient = complex(iop[p, q])
+            if coefficient and p >= q:
+                qubit_operator += coefficient * bksf_one_body(edge_matrix, p, q)
+
+            # Keep looping for the two-body terms.
+            for r in range(n_qubits):
+                for s in range(n_qubits):
+                    coefficient = complex(iop[p, q, r, s])
+
+                    # Skip zero terms.
+                    if (not coefficient) or (p == q) or (r == s):
+                        continue
+
+                    # Identify and skip one of the complex conjugates.
+                    if [p, q, r, s] != [s, r, q, p]:
+                        if len(set([p, q, r, s])) == 4:
+                            if min(r, s) < min(p, q):
+                                continue
+                        elif p != r and q < p:
+                                continue
+
+                    # Handle the two-body terms.
+                    transformed_term = bksf_two_body(edge_matrix,p, q, r, s)
+                    transformed_term *= coefficient
+                    qubit_operator += transformed_term
+    return qubit_operator
+
+def bksf_edge_matrix(iop,n_qubits=None):
```

Comment:
again, no acronyms. Also, you need to have a space between "iop," and "n_qubits=None". You should use the pep8 linter, which I think would point out little stupid spacing style issues like this.

Install the pep8 linter. Then open the file:
~/.config/pep8 and put the following inside:
[pep8]
ignore = E111, E114, E226

Then, before you commit code please run pep8 *.py.

This should point out such issues.